### DISCUSSION TITLE
Ignore file in node_modules

### DISCUSSION THREAD

**Discussion by IvanRomanovski** [babel/babel#12231] [Q&A] [answered]
I have a large 4 MB JS file in a node_module (result of compilation from C++ to asm.js). I am trying to include it in a react native project and it works but babel parsing takes a long time. I want to skip babel parsing for this single js file. I have tried ignoring it in .babelrc but the ignore seems to be ignored.

If I understand 7.x. .babelrc changes correctly (see below), .babelrc is not applied to node_modules anymore. Is there any way around this?

> **6.x vs 7.x .babelrc loading**
> Users coming from Babel 6.x will likely trip up on these two edge cases, which are new in Babel 7.x. These two restrictions were added to address common footguns in Babel 6.x:
> 
> - .babelrc files applied to node_modules dependencies, often unexpectedly.
> - .babelrc files failed to apply to symlinked node_modules when people expected them to behave like normal dependencies.
> - .babelrc files in node_modules dependencies would be detected, even though the plugins and presets inside they were generally not installed, and may not even be valid in the version of Babel compiling the file.

**Comment by nicolo-ribaudo** [comment, marked as answer]
You can use a `babel.config.json` (or `.js`), which is applied to the whole project (with dependencies) rather than to the single package.

**Reply by IvanRomanovski** [reply]
Thanks. I have tried to modify my `babel.config.js` like so:
```
module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  exclude: [
    /(mylargelib\.)./,
  ]
};
```

However it didn't seem to influence anything. Is my syntax correct? If it is it must be metro bundler that is ignoring this...

**Reply by nicolo-ribaudo** [reply]
[`exclude`](https://babeljs.io/docs/en/options#exclude) means "I don't want to apply this config portion to this files", which means that the large lib is still being compiled but without `metro-react-native-babel-preset`. This can be useful where you are configuring Babel in multiple points (for example, `babel.config.js` + `babel-loader`) options, because it will still enable the plugins defined in `babel-loader`'s options.
In your case this probably means that Babel is parsing that large file, not transforming it (because there are no plugins/presets) enabled, and re-converting it to a string.

The option you need is [`ignore`](https://babeljs.io/docs/en/options#ignore), which tells Babel not to process a given file:
```js
module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  ignore: [
    /(mylargelib\.)./,
  ]
};
```

**Reply by IvanRomanovski** [reply]
Thanks. This answers my question.

Unfortunately, after logging out which files cause the transformation to hang, it seems that metro bundler is running some babel transformations on final concatenated bundle, which I can't exclude from babel config. (For any other poor soul reading this - `NODE_OPTIONS=--max_old_space_size=8192 yarn start` seems to be the only way to bundle large JS files in React Native),
